src/state/subscribers/logic.js

```diff
@@ -0,0 +1,70 @@
+import { createLogic } from "redux-logic"
+import {
+  map,
+} from "lodash";
+
+import { 
+  REQUEST_FAILED,
+  SUBMIT_SUBSCRIBER_SUCCESS,
+  SUBMIT_SUBSCRIBER,
+  REQUEST_ALL_SUBSCRIBERS,
+  REQUEST_ALL_SUBSCRIBERS_SUCCESS,
+} from "./constants";
+
+const submitSubscriberLogic = createLogic({
+    process({
+        firebasedb, 
+        action,
+      }) {
+      const {
+        payload,
+      } = action;
+      let subKey = '';
+      if (payload.key) {
+        subKey = payload.key;
+      } else {
+        subKey = firebasedb.ref(`subscribers/`).push().key;
+      }
+      delete payload.key;
```

Comment:
instead of editing payload directly here, consider making a new object, `{...payload,  key: null}`

Reply:
So what I'm wanting to do here is not set the 'key' attribute to null but remove it altogether before updating firebase. Originally it wasn't part of the Subscriber object in firebase, but I added it to the form in a hidden field to serve as an ID for the subscriber (Without it, whichever field I use to identify the subscriber would have to be made non-editable to prevent unexpected behavior, and I  think we want email, name, and districts to all be editable).

If I'm breaking a convention by editing the payload, I could pull just the email, name, districts attributes out of the payload and make a new object for updating firebase, leaving the .then() statement as-is. Would that be better?

Reply:
If a field is `null` firebase won't save it at all

Reply:
OK, I see. That's cleaner, I like it, thanks!